---
title: "PPG II: An open, community-derived classification for ferns and lycophytes"
format: docx
bibliography: references.yaml
execute: 
  echo: false
  message: false
  warning: false
---

```{r}
#| label: setup
source(here::here("R/packages.R"))
source(here::here("R/functions.R"))
library(glue)
library(tidyverse)
library(assertr)
library(assertthat)
library(taxlist)
library(taxnames)
library(english)
library(knitr)
library(ggtree)
library(ggrepel)
library(ftolr)
library(patchwork)

conflicted::conflict_prefer("filter", "dplyr")

# Load targets
tar_load(
  c(
    commenters,
    children_tally,
    issues_count_plot,
    ppg_issues,
    ppg_issues_count,
    ppg_i_taxa_count,
    ppg_2_taxa_count,
    voting_results,
    tree_fig,
    ppg_classification
  ),
  store = here::here("_targets")
)
```

Running title: PPG II

The Pteridophyte Phylogeny Group

```{r}
#| label: load-names

authors <- read_csv("data/ppg2_author_list.csv") |>
  filter(str_detect(name, "Nitta|Schuettpelz|Schneider|Rothfels", negate = TRUE)) |>
  mutate(
    last_name = map_chr(name, extract_last_name)
  ) |>
  arrange(last_name) %>%
  mutate(auth_num = seq(5, nrow(.) + 4, by = 1)) |>
  mutate(
    institution = tidyr::replace_na(institution, "?"),
    country = tidyr::replace_na(country, "?"),
    name_print = glue::glue("{name}^{auth_num}^"),
    inst_print = glue::glue("^{auth_num}^ {institution}, {country}")
  )

other_authors <- paste(authors$name_print, collapse = ", ")
institutions <- paste(authors$inst_print, collapse = "\n")

```

Recommended citation: PPG II (2026). This project was organized by Joel H. Nitta\*^1^, Eric Schuettpelz\*^2^, Harald Schneider^3^, Carl Rothfels^4^; with additional contributions from (in alphabetical order): `r other_authors`

`r institutions`

\*Authors for correspondence

```{r}
#| label: abstract

# Format ppg2 taxon numbers for printing
n <- ppg_2_taxa_count |>
  select(taxonRank, ppg2 = accepted) |>
  left_join(
    select(ppg_i_taxa_count, taxonRank, ppg1 = accepted),
    by = "taxonRank"
  ) |>
  pivot_longer(
    names_to = "system",
    values_to = "n",
    c(ppg1, ppg2)
  ) |>
  mutate(
    n = case_when(
      n < 10 ~ as.character(english::english(n)),
      n >= 10 ~ scales::number(n, big.mark = ",")
    )
  ) %>%
  pivot_wider(names_from = system, values_from = n) %>%
  split(.$taxonRank)
```

**Abstract** 

The taxonomy of ferns and lycophytes---the paraphyletic group known as the pteridophytes---has undergone several major revisions as knowledge about the phylogenetic relationships of these plants has accumulated.
The most recent widely accepted classification for ferns and lycophytes is Pteridophyte Phylogeny Group (PPG) I, published in 2016.
Since that time, many new names have been published and phylogenetic relationships come into clearer focus.
Here, we present PPG II, an updated classification to reflect advances made in pteridophyte systematics since PPG I.
PPG II recognizes `r n$class$ppg2` classes, `r n$order$ppg2` orders, `r n$family$ppg2` families, `r n$genus$ppg2` genera, and `r n$species$ppg2` species of ferns and lycophytes.
Like PPG I, PPG II is the result of decisions made by a diverse community of taxonomic experts, but improvements have been made to increase transparency and efficiency, including posting all discussions on an openly available website and voting on taxonomic proposals on a rolling, monthly basis.
PPG II also extends the classification to the species level and includes nothogenera, neither of which were treated in PPG I.
Going forward, incremental updates to PPG II will be made online, obviating the lengthy delay between updated treatments.
The PPG II classification can be accessed on our website (<https://pteridogroup.github.io/ppg>) or World Flora Online (<https://wfoplantlist.org>).

**Key words:** classification, ferns, lycophytes, open science, phylogeny, pteridophytes, taxonomy, World Flora Online

{{< pagebreak >}}

Ferns and lycophytes--comprising the paraphyletic pteridophytes--are globally important groups of plants known for their resilience during their long phylogenetic history [@li2025fernsa].
Our understanding of fern and lycophyte relationships was revolutionized by DNA sequencing, which revealed that ferns are sister to seed plants and only distantly related to the lycophytes (thus "pteridophytes" are not monophyletic), and that horsetails and whisk ferns are nested within ferns [@pryer2001horsetails].
The first major revised classification of the molecular-phylogenetic era was @smith2006classification, which was limited to taxa at the rank of family and higher.
Two subsequent classifications extended the taxonomic scope to genera [@christenhusz2011linear; @christenhusz2014trends], but due to their drastic consolidation of genera and families they were not widely accepted by the broader pteridophyte community [@schuettpelz2018are].
The next widely accepted classification was Pteridophyte Phylogeny Group (PPG) I [-@pteridophytephylogenygroupi2016communityderived].
PPG I was remarkable for its community-based approach, which sought to obtain consensus among fern and lycophyte experts by including nearly 90 co-authors.

Since the publication of PPG I, phylogenetic relationships have been further clarified and taxonomic novelties have accrued.
Accordingly, the PPG community here puts forth a revised taxonomic system for ferns and lycophytes, PPG v2.0.0 (equivalently, PPG II).

## Methods

To enable maximal participation in taxonomic discussions and ensure that the reasoning behind each taxonomic decision is openly available, we implemented a system that is, to our knowledge, unique amongst taxonomic communities.
We briefly discuss this system below; further technical details will be the focus of another publication.

### Taxonomic criteria

Our general criteria for approving taxonomic proposals remained unchanged from PPG I [-@pteridophytephylogenygroupi2016communityderived]: we sought to only recognize monophyletic groups (clades), while minimizing taxonomic disruption and maximizing usefulness (diagnosability).
Additionally proposals must: (1) implement a change relative to PPG I [-@pteridophytephylogenygroupi2016communityderived]; (2) only include names that have already been validly published under the International Code for Algae, Fungi, and Plants (hereafter, “the Code”); (3) involve a taxon or taxa at the genus level or higher.
The one exception to the second rule is in the case where nomenclatural conservation proposals are before the International Committee on Plant Nomenclature (ICPN).
Because such proposals may not be voted on by the ICPN for several years, we allowed such names to be voted on and approved by PPG under the condition that a nomenclatural proposal had already been submitted.
In the case that the nomenclatural proposal is rejected, we will update the PPG classification accordingly.

### Discussion forum and voting

We set up a GitHub repository (<https://www.github.com/pteridogroup/ppg>) to solicit taxonomic proposals from the PPG community.
We used GitHub's "issue" functionality to track and discuss proposals; GitHub issues are normally used to track bugs during software development, but we found they also work well for managing taxonomic discussions.
We designed a custom issue template to ensure that all proposals included standard information such as the proposed taxon, (approximate) number of affected species, reason for the change, and citations of relevant literature.
Anyone could submit a proposal providing they met the taxonomic criteria and agreed to our Code of Conduct; proposals were then subject to an open discussion period of at least one month.
During this time, anyone who agreed to our Code of Conduct could comment on the proposal.

Following the discussion period, voting on proposals was conducted via Google Forms.
For each proposal, PPG members could choose to agree, disagree, or abstain.
Proposals receiving 2/3 agreement passed and were recognized by PPG.
All taxonomic discussions and results of voting are available for anybody to access on our GitHub repository.
The discussion and voting process ran continuously, with a ballot issued monthly.

### Database management

To maintain taxonomic data, PPG uses World Flora Online (WFO), a large-scale database project to provide a consensus taxonomy for all plants (<https://wfoplantlist.org>). 
World Flora Online relies on a network of taxonomic experts, called Taxonomic Expert Networks (TENs) to curate taxonomy of various plant groups. 
PPG is the TEN for ferns and lycophytes.
As such, all taxonomic decisions made by PPG also become the taxonomy available in WFO.

The database editor provided by WFO, called Rhakhis (<https://list.worldfloraonline.org/rhakhis/ui>), is taxonomy-aware---that is, it enforces taxonomic rules, for example, that synonyms must map on to an accepted name and not another synonym---and provides full functionality for tracking synonymy relationships, publication and author data, and nomenclatural status, among other taxonomically relevant information.
Furthermore, Rhakhis is available as a live, web-based database, so PPG members can collaboratively edit the data in real-time, with no need for synchronizing database files across multiple machines.

The initial version of the taxonomic data was based on a copy of the World Ferns database provided by Michael Hassler (<https://www.worldplants.de/world-ferns/ferns-and-lycophytes-list>), which was imported into Rhakhis and edited to reflect taxonomic proposals that passed voting.

### Extension to the species level

Another major change from PPG I is that the PPG II classification includes names at the rank of species and below (PPG I extended only to genus).
By starting from the World Ferns classification (<https://www.worldplants.de/world-ferns/ferns-and-lycophytes-list>), PPG II was "pre-populated" with these lower ranked taxa.
As changes to higher taxa were approved through our voting process, we used Rhakhis to remap taxa at the level of species and below to the appropriate position in the revised classification.
However, since the goal of the current project was to revise the genus level and higher classification, we have not inspected all species names, nor is this feasible to carry out by discussing and voting as was done for names at higher taxonomic levels due to the large number of species (ca. 14,000 accepted names).
Therefore, we have established committees of specialists who are each responsible for maintaining taxonomic data at the species level and below in Rhakhis for particular groups within pteridophytes (typically the genus or family level).
This work will continue indefinitely with the publication of PPG II.

### Versioning

Starting with this version of the PPG classification (PPG 2.0.0), we are implementing a semantic versioning system, where the smallest digit indicates changes to the taxonomy at the species-level or below, the middle digit indicates changes at genus-level or higher, and the largest digit indicates major changes (major rearrangements of the taxonomic hierarchy).
New versions will be issued as needed, for example following recognition of a new genus or if a specific version is needed to be cited by a project like the Fern Tree of Life [@nitta2022open].
We anticipate that updates (new versions) will be published much more frequently than the time that has elapsed between PPG I and PPG II.
This will greatly benefit stakeholders who need an up-to-date, standardized taxonomy for ferns and lycophytes, while simultaneously providing, through the version system, the ability to unambiguously reference a specific snapshot of the classification.

## Results

```{r}
#| label: proposal-results
#| include: false

# Check that analysis of voting results matches issue status extracted
# from issue header
ppg_issues |>
  select(issue = number, state, status) |>
  left_join(
    voting_results,
    by = join_by(issue),
    relationship = "one-to-one"
  ) |>
  mutate(
    result = str_replace_all(result, "passes", "PASSED") |>
      str_replace_all("does not pass", "NOT PASSED"),
    result = tidyr::replace_na(result, "TBA")
  ) |>
  verify(all.equal(status, result), success_fun = success_logical)

# Summarize time period of submitted issues
# For month, add period only if abbreviated (not for May)
months_abb <- as.character(month(1:12, label = TRUE, abbr = TRUE))
months_full <- as.character(month(1:12, label = TRUE, abbr = FALSE))
months_abb_punc <- ifelse(months_abb == months_full, months_abb, paste0(months_abb, "."))

issues_time <-
  ppg_issues |>
  select(created_at) |>
  mutate(created_at = lubridate::ymd_hms(created_at)) |>
  summarize(
    min = min(created_at),
    max = max(created_at)
  ) |>
  mutate(
    min_year = year(min),
    min_month = months_abb_punc[month(min)],
    max_year = year(max),
    max_month = months_abb_punc[month(max)]
  )

# Count number of commenters per valid issue
comment_count <- commenters |>
  filter(!is.na(commenter)) |>
  group_by(issue) |>
  mutate(n_commenters = n_distinct(commenter)) |>
  # don't count if joelnitta was the only commenter
  # (standard announcement, not discussion)
  filter(!(commenter == "joelnitta" & n_commenters == 1)) |>
  ungroup() |>
  select(issue, n_commenters) |>
  unique() |>
  arrange(desc(n_commenters))

# Summarize voting stastics
n_issues_passed <- ppg_issues |>
  filter(str_detect(title, "\\[PASSED\\]")) |>
  nrow()

vote_stats <-
  voting_results |>
  filter(!is.na(result)) |>
  summarize(
    mean = round(mean(total_votes), 1),
    min = min(total_votes),
    max = max(total_votes)
  )

```

A total of `r nrow(ppg_issues)` valid proposals were made from `r issues_time$min_year` `r issues_time$min_month` to `r issues_time$max_year` `r issues_time$max_month`, of which `r n_issues_passed` were accepted and adopted into PPG II (@fig-voting).
Proposals averaged `r vote_stats$mean` votes (range `r vote_stats$min` to `r vote_stats$max` votes), and `r nrow(comment_count)` proposals featured a discussion among PPG members beyond the initial proposal itself or standard announcements such as voting results.

```{r}
#| label: results
#| include: false

# make sure n class, subclass, order, and suborder agree between
# ppg I and ppg II
ppg_2_higher <-
  ppg_2_taxa_count |>
  filter(taxonRank %in% c("class", "subclass", "order", "suborder")) |>
  select(taxonRank, accepted)

ppg_i_higher <-
  ppg_i_taxa_count |>
  filter(taxonRank %in% c("class", "subclass", "order", "suborder"))

assert_that(
  magrittr::equals(
    nrow(
      anti_join(ppg_i_higher, ppg_2_higher, by = join_by(taxonRank, accepted))
    ),
    0
  ),
  msg = "Number of higher taxa in PPG I and PPG II don't match"
)
```

The higher taxonomic levels of PPG II are largely unchanged relative to PPG I.
Two classes of pteridophytes are recognized, Lycopodiopsida (lycophytes) and Polypodiopsida (ferns) (@fig-tree).
These two classes are distinct lineages; ferns are more closely related to seed plants than they are to lycophytes [@kenrick1997origin; @pryer2004phylogeny; @schneider2004ferns].
However, lycophytes and ferns have historically been studied together as seed-free vascular plants and we continue in that tradition.
Our recognition here of `r n$subclass$ppg2` subclasses, `r n$order$ppg2` orders, and `r n$suborder$ppg2` suborders is also unchanged from PPG I.

PPG II begins to diverge from PPG I at the level of family, where `r n$family$ppg2` families are recognized versus `r n$family$ppg1` in PPG I (@tbl-taxa).
The number of subfamilies and genera are also both greater in PPG II vs. PPG I (`r n$subfamily$ppg2` vs. `r n$subfamily$ppg1` and `r n$genus$ppg2` vs. `r n$genus$ppg1`, respectively).
Most taxonomic changes come via the addition of new names, i.e., splitting (@tbl-changes), whereas a smaller number of changes stems from the subsumption of existing names, i.e., lumping (@tbl-changes).
Another change is the recognition of `r n$nothogenus$ppg2` nothogenera, which were not treated in PPG I [-@pteridophytephylogenygroupi2016communityderived].
The number of taxa at the level of species and below (PPG II recognizes approximately 14,000 accepted species) precludes publishing those names here, but the full classification (extending to subspecific ranks) is available on our website (<https://pteridogroup.github.io/>) and WFO (<https://www.worldfloraonline.org/>).

## Discussion

### An open, community-based approach to taxonomy

PPG is unique in its open, community-based approach to taxonomy.
While other community efforts have been established [e.g., Legume Phylogeny Working Group; @leroux2022experta] to our knowledge, no other, no comparable classification working group exists with a similar number of members that makes its discussions and results completely transparent and publicly available.
In addition, the non-hierarchical, inclusive nature of our approach reduces the risk of our taxonomic decisions being driven by a vocal minority. 
It is our hope that this commitment to transparency and accessibility has resulted, and will continue to result, in a high-quality classification that is widely used by stakeholders spanning a broad range of backgrounds, including researchers, conservationists, local botanists, curators, teachers, and others.

One clear benefit of the community approach is the ability to keep up with the rapid pace of systematic research.
We live in an age where the generation of data far exceeds the ability of any single individual to comprehensively track new insights across such a large and heterogeneous group as pteridophytes.
The community approach allows us to distribute the workload and stay abreast of the latest advances in the systematics of these plants, which leads to a continuously updated, maximally useful taxonomy.

### Trends in pteridophyte classification

```{r}
nchild <- children_tally %>%
  split(.$taxon_name)
```

The general trend has been an increase in genus number.
This is driven primarily by a few (sub)families: Thelypteridaceae (`r nchild$Thelypteridaceae$n_genera` vs. 30), Grammitidoideae (`r nchild$Grammitidoideae$n_genera` vs. 33), and Blechnaceae (`r nchild$Blechnaceae$n_genera` vs. 18; each comparison is number of genera recognized in PPG II vs. PPG I).
Notable exceptions to the trend of splitting include recognition of a broadly construed *Lepisorus*, subsuming six genera that were recognized in PPG I.

### Next steps

The vast majority of taxa recognized herein are well established as monophyletic and were supported by a strong majority of voting members.
However, there are still areas of the classification that need resolution, and in some case we recognize non-monophyletic taxa because no clear monophyletic option exists.
Known non-monophyletic taxa have been noted in the classification.
In some cases, relatively deep relationships have still not been adequately resolved, especially due to conflicts between nuclear and plastid datasets, as indicated by the dashed lines in @fig-tree and summarized in @tbl-uncertain.

One advantage of our approach using rolling, monthly voting is that, going forward, a taxonomic proposal may be submitted at any time, and proposals that did not pass may be revised and/or revisited in the future.
Furthermore, the classification can be updated and circulated online with much greater frequency than possible under a traditional print-based model whereby only major updates are published.
This flexibility allows us to revisit taxonomic areas of uncertainty and revise them as soon as new data are available.

## Classification

Circumscription and synonymy are not shown; this information can be found on the digital version of the classification at <https://pteridogroup.github.io> or at World Flora Online.
Species counts should be considered approximate.
All taxa are considered monophyletic unless otherwise mentioned.

```{r}
#| label: ppg-treatment
#| results: asis

cat(ppg_classification, sep = "\n")
```

## Tables

```{r}
#| label: tbl-taxa
#| tbl-cap: "Number of taxa in PPG II vs. PPG I by taxonomic status
#|   (accepted or synonym). Complete synonymy was not presented in PPG I
#|   so this is not shown. PPG I did not treat nothotaxa."

ppg_2_taxa_count |>
  left_join(
    select(
      ppg_i_taxa_count,
      taxonRank,
      ppgi_accepted = accepted
    )
  ) |>
  filter(taxonRank != "tribe") |>
  mutate(
    taxonRank = str_to_sentence(taxonRank),
    across(where(is.integer), ~ scales::number(., big.mark = ","))
  ) |>
  select(
    Rank = taxonRank,
    `PPG II (accepted)` = accepted,
    `PPG II (synonym)` = synonym,
    `PPG I (accepted)` = ppgi_accepted
  ) |>
  knitr::kable()
```

```{r}
#| label: tbl-changes
#| tbl-cap: "Count of taxonomic changes in PPG II vs. PPG I by rank, with
#|   the effect on the total number of taxa in each rank brought about by each
#|   change. Types of change include conserve: conserves one
#|   name over another; sink: subsumes one taxon into another; split: recognizes
#|   a new taxon; transfer: moves a taxon into a different position in the
#|   hierarchy."

ppg_issues_count |>
  filter(change != "none") |>
  mutate(rank = factor(rank, levels = c("Family", "Subfamily", "Genus", "Nothogenus"))) |>
  count(rank, change) |>
  mutate(effect = case_when(
    change %in% c("conserve", "expand", "shrink", "transfer") ~ "none",
    change == "sink" ~ "decrease",
    change == "split" ~ "increase",
  )) |>
  filter(
    change %in% c("conserve", "sink", "split", "transfer")
  ) |>
  mutate(
    change = str_to_sentence(change),
    effect = str_to_sentence(effect)
  ) |>
  arrange(rank, change) |>
  select(
    Rank = rank,
    Change = change,
    Effect = effect,
    n = n
  ) |>
  knitr::kable()
```

```{r}
#| label: tbl-uncertain
#| tbl-cap: "Uncertain relationships in pteridophytes. Node indicates numbered
#|   nodes in Fig. 1."
read_csv("data/table_uncertainty.csv") |>
  knitr::kable()
```

## Figures

```{r}
#| label: fig-voting
#| fig-width: 6.5
#| fig-height: 5
#| fig-cap: "Number (A) and results of voting (B) on taxonomic proposals
#|   submitted by PPG community members for PPG II."
ggsave(
  plot = issues_count_plot,
  width = 6.5,
  height = 5,
  units = "in",
  file = "fig_1.pdf"
)

issues_count_plot
```

```{r}
#| label: fig-tree-caption

fig_tree_cap <- glue::glue("
  Summary cladogram showing relationships between families of tracheophytes recognized by PPG II.
  Topology for Polypodiopsida based on Fern Tree of Life v{ft_data_ver()}, which was inferred using plastid data.
  Numbers next to each family indicate number of genera followed by number of species.
  Diamonds at tips indicate relative number of species in each family.
  Numbers at nodes indicate uncertain relationships, summarized in Table 3.
  ") |>
  as.character() |>
  str_remove_all("//\n") |>
  str_squish()

```

```{r}
#| label: fig-tree
#| fig-width: 6.5
#| fig-height: 9
#| fig-cap: !expr 'fig_tree_cap'

ggsave(
  plot = tree_fig,
  width = 6.5,
  height = 8.5,
  units = "in",
  file = "fig_2.pdf"
)

tree_fig
```

## References
