---
title: "PPG II: An open, community-derived classification for ferns and lycophytes"
format: docx
bibliography: references.yaml
execute: 
  echo: false
  message: false
  warning: false
---

```{r}
#| label: setup
source(here::here("R/packages.R"))
source(here::here("R/functions.R"))
library(glue)
library(tidyverse)
library(assertr)
library(assertthat)
library(taxlist)
library(taxnames)
library(english)
library(knitr)
library(ggtree)
library(ggrepel)
library(ftolr)

conflicted::conflict_prefer("filter", "dplyr")

# Load targets
tar_load(
  c(
    ppg_tl,
    ppg,
    comments,
    families_in_phy_order,
    children_tally,
    ppg_issues_count,
    ppg_i_taxa_count,
    ppg_2_taxa_count,
    phy_family
  ),
  store = here::here("_targets")
)
```

Running title: PPG II

The Pteridophyte Phylogeny Group

Recommended citation: PPG II (2026). This project was organized by Eric Schuettpelz\*^1^, Joel H. Nitta\*^2^, Harald Schneider^3^, Carl Rothfels^4^, ...

\*Authors for correspondence

```{r}
#| label: abstract

# Format ppg2 taxon numbers for printing
n <- ppg_2_taxa_count |>
  select(taxonRank, ppg2 = accepted) |>
  left_join(
    select(ppg_i_taxa_count, taxonRank, ppg1 = accepted),
    by = "taxonRank"
  ) |>
  pivot_longer(
    names_to = "system",
    values_to = "n",
    c(ppg1, ppg2)
  ) |>
  mutate(
    n = case_when(
      n < 10 ~ as.character(english::english(n)),
      n >= 10 ~ scales::number(n, big.mark = ",")
    )
  ) %>%
  pivot_wider(names_from = system, values_from = n) %>%
  split(.$taxonRank)
```

**Abstract** 

The taxonomy of ferns and lycophytes--the paraphyletic group known as the pteridophytes--has undergone several major revisions as knowledge of the phylogenetic relationships of these plants accumulates.
The most recent widely accepted taxonomy for ferns and lycophytes is Pteridophyte Phylogeny Group I, published in 2016.
Since that time, many new names have been published and the relationships of various taxa have come into clearer focus.
Here, we present a new taxonomy that reflects advances made in pteridophyte systematics since PPG I.
PPG II recognizes `r n$class$ppg2` classes, `r n$order$ppg2` orders, `r n$family$ppg2` families, `r n$genus$ppg2` genera, and `r n$species$ppg2` species of ferns and lycophytes.
Like PPG I, PPG II is the result of decisions made by a diverse community of taxonomic experts, but we have made improvements to the system to increase transparency and efficiency, including posting all discussions on an openly available website and voting on taxonomic proposals on a rolling, monthly basis.
Furthermore, PPG II includes names to the species level and nothogenera, neither of which were treated in PPG I.
Updates to PPG II will be made on a much more frequent basis by using online resources, obviating the lengthy delay between updated treatments.
PPG II can be accessed on our website (<https://pteridogroup.github.io/ppg>) or World Flora Online (<https://wfoplantlist.org>).

**Key words:** classification, ferns, lycophytes, open science, phylogeny, pteridophytes, taxonomy

##

Ferns and lycophytes--forming together the paraphyletic pteridophytes--are globally important groups of plants known for their resilience during their long phylogenetic history. 
Early taxonomic classification of pteridophytes relied on macroscopic characteristics, in particular, the form and arrangement of sporangia and spores.
After Darwin’s theory of evolution became widely recognized, taxonomists sought to recognize natural groups, which was subsequently facilitated by the introduction of phylogenetic analyses to identify clades.
Understanding of the phylogentic relationships of ferns and lycophytes were revolutionized by DNA sequencing, starting with the plastid *rbcL* gene.
In particular, DNA analysis revealed that ferns are sister to seed plants (thus "pteridophytes" are not a natural group) and that horsetails and whisk ferns are not merely "fern allies" but nested within ferns [@pryer2001horsetails].
The first major revised classification of the molecular-phylogenetic era was @smith2006classification, but this only treated taxa at the level of family and higher.
The next major taxonomy was Pteridophyte Phylogeny Group (PPG) I [-@pteridophytephylogenygroupi2016communityderived], which extended the taxonomic scope to genera.
PPG I was remarkable for its expert community-based approach, which sought to obtain consensus among fern and lycophyte experts by including nearly 90 co-authors.
Since the publication of PPG I, taxonomic novelties and clarification of phylogenetic relationships have significantly accrued.
Accordingly, the PPG community here puts forth a revised taxonomic system for ferns and lycophytes, PPG v2.0 (equivalently, PPG II).

## Methods

To enable maximal participation in taxonomic discussions and ensure that the reasons for each taxonomic decision are openly available, we have implemented a system that is to our knowledge unique amongst taxonomic communities.
We briefly discuss this below.
Further technical details will be made available in another publication.

### Discussion forum and voting

We set up a GitHub repo (<https://www.github.com/pteridogroup/ppg>) to solicit taxonomic proposals from the PPG community.
We used GitHub's "issue" functionality to track and discuss proposals; GitHub issues are normally used to track bugs during software development, but we have found they work well for managing taxonomic discussions as well.
We designed a custom issue template to ensure that all proposals include standard information such as the proposed taxon, (approximate) number of affected species, reason for the change, and citations of relevant literature.
Anybody may submit a proposal providing they agree to our Code of Conduct and the proposal meets the following criteria: the proposal must be change relative to PPG I (2016); the proposal must only include names that have already been validly published under the International Code for Algae, Fungi, and Plants (hereafter, "the Code"); the proposal is a change to name at the genus level or higher.
Proposals meeting the criteria for inclusion are then subject to a discussion period of at least one month.
During this time, anybody who agrees to our Code of Conduct may comment on the proposal.
Our criteria for approving taxonomic proposals remains unchanged from PPG I (2016): we seek to only recognize monophyletic groups (clades), while minimizing taxonomic disruption and maximizing usefulness (diagnosability).

Following the discussion period, voting on proposals is conducted via Google Forms.
For each proposal, PPG members may choose to agree, disagree, or abstain.
Proposals receiving 2/3 agreement pass and are recognized by PPG.
All taxonomic discussions and results of voting are available for anybody to access on our GitHub repo.
The discussion and voting process runs continuously, with a ballot issued monthly.

### Database management

To maintain taxonomic data, PPG uses a database editor provided by World Flora Online (WFO), a large-scale project to provide a consensus taxonomy for all plants (<https://wfoplantlist.org>). 
World Flora Online relies on a network of taxonomic experts, called Taxonomic Expert Networks (TENs) to curate taxonomy of various plant groups. 
PPG is the TEN for ferns and lycophytes.
As such, all taxonomic decisions made by PPG also become the taxonomy used by WFO.

The database editor provided by WFO called Rhakhis () is taxonomy-aware; that is, it enforces taxonomic rules, for example, that synonyms must map on to an accepted name and not another synonym.
Furthermore, Rhakhis is available as a live, web-based database.
This means that PPG members can collaboratively edit the data in real-time, with no need for synchronizing database files across multiple machines.

The initial version of the database a copy of the World Ferns database provided by Michael Hassler (<https://www.worldplants.de/world-ferns/ferns-and-lycophytes-list>). 
It was imported into Rhakhis and edited to reflect taxonomic proposals that passed voting.

## Results

```{r}
#| label: results

# make sure n class, subclass, order, and suborder agree between
# ppg I and ppg II
ppg_2_higher <-
  ppg_2_taxa_count |>
  filter(taxonRank %in% c("class", "subclass", "order", "suborder")) |>
  select(taxonRank, accepted)

ppg_i_higher <-
  ppg_i_taxa_count |>
  filter(taxonRank %in% c("class", "subclass", "order", "suborder"))

assert_that(
  magrittr::equals(
    nrow(
      anti_join(ppg_i_higher, ppg_2_higher, by = join_by(taxonRank, accepted))
    ),
    0
  ),
  msg = "Number of higher taxa in PPG I and PPG II don't match"
)
```

The higher taxonomic levels of PPG II are largely unchanged compared to PPG I.
Two classes of pteridophytes were recognized, Lycopodiopsida (lycophytes) and Polypodiopsida (ferns) (@fig-tree).
These two classes are distinct lineages; ferns are actually more closely related to seed plants than they are to lycophytes.
However, lycophytes and ferns have historically been studied together as seed-free vascular plants and we continue in that tradition.
Recognition of `r n$subclass$ppg2` subclasses, `r n$order$ppg2` orders, and `r n$suborder$ppg2` suborders is also the same between PPG I and II.

Changes to PPG II become apparent at the level of family, where `r n$family$ppg2` families are recognized versus `r n$family$ppg1` in PPG I (@tbl-taxa).
The number of subfamilies and genera are also greater in PPG II vs. PPG I (`r n$subfamily$ppg2` vs. `r n$subfamily$ppg1` and `r n$genus$ppg2` vs. `r n$genus$ppg1`, respectively).
Most taxonomic changes are additions of new names, i.e., splitting (@tbl-changes).
A smaller number of changes are subsumption of existing names, i.e., lumping (@tbl-changes).
Another change is the recognition of `r n$nothogenus$ppg2` nothogenera, which were not treated at all in PPG I [-@pteridophytephylogenygroupi2016communityderived].

## Discussion

### An open, community-based approach to taxonomy

PPG is unique in its open, community-based approach to taxonomy.
While other community efforts have been established (e.g. Legume Phylogeny Working Group), to our knowledge, no other taxonomic group exists with a similar number of members that makes its discussions and results completely transparent and available.
It is our hope that this approach has, and will continue to, result in a high-quality taxonomy that is widely used by stakeholders spanning a broad range of backgrounds, including researchers, conservationists, local botanists, curators, teachers, and others.
This is not to say that the community approach is not without shortcomings.
In some cases, unilateral decisions may result in a more standardized system such as the evolutionary depth at which to recognize various taxa above the rank of species.
However, during the discussion period, we have frequently seen participants engaging in discussions where the proposed circumscription is compared with previous cases and the arguments presented for and against the proposed change.
We are confident that ours is a self-correcting system that ultimately results in a taxonomy that most closely aligns with consensus expert opinion, and is therefore worthwhile to be used by the broader botanical community.
One clear benefit of the community approach is the ability to keep up with the rapid pace of systematic research.
We live in an age where the generation of data far exceeds the ability of any single individual to comprehensively track such a large and heterogeneous group as pteridophytes.
The community approach allows us to distribute the workload and stay abreast of the latest advances in the systematics of these plants, which leads to a continuously updated, maximally useful taxonomy.

### Areas of uncertainty

There are still parts of the pteridophyte taxonomy that need resolution, or at least improvements to overcome ambiguities.
One complicated case concerns *Lecanopteris* and related lineages.
Recent phylogenies show that the myrmecophilous *Lecanopteris* s.s. is nested within a clade that includes taxa formerly placed in *Microsorum*, but that this clade is distinct from the lineage containing the type species of *Microsorum* [@chen2020exploring; @perrie2021expansion].
During voting, multiple competing proposals were put forward, including recognition of *Lecanopteris* s.s. and recognition of separate genera for each of the former *Microsorum* lineages, as well as a broadly circumscribed *Lecanopteris* that would subsume the former *Microsorum* lineages.
Ultimately, *Lecanopteris* sensu lato was rejected, but only one of the segregate generea, *Bosmania* was recognized, leaving *Microsorum* paraphyletic.
We anticipate that further systematic studies of this group will provide more robust results that will eventually allow for a recognition of monophyletic genera.
While a complete description of every area of taxonomic uncertainty is beyond the scope of this paper, a summary may be found in Table 3.

One advantage of the current approach is that a taxonomic proposal may submitted at any time, and proposals that did not pass may be revisited in the future.
Furthermore, the taxonomy can be updated and circulated online (see Next Steps) with much greater frequency than possible under a model of only publishing each version as a scientific paper.
This allows us to revisit taxonomic areas of uncertainty and revise them as soon as new data come to light.

While the continuously updated online approach allows us to make incremental updates, this has presented a challenge vis-a-vis the model used by the International Committee on Plant Nomenclature, which only votes on proposals to conserve names on a much more infrequent basis.
Rather than requiring every taxonomic treatment needing nomenclatural approval to wait on a vote by the nomenclatural committee, which could delay its usage by several years, we have established the rule that such names may be voted on an approved by PPG as long as a nomenclatural proposal has already been submitted.
In the unusual case that the nomenclatural proposal is rejected, PPG would also then reject using the name.

### Next steps

A major difference in approach between PPG I and PPG II is that further updates to PPG II will be made online instead as a full scientific paper.
Starting with PPG II (PPG 2.0.0), we are implementing a semantic versioning system, where the smallest digit indicates changes to the taxonomy at species or below, the middle digit indicates changes at the level of genus to family, and the largest digit indicates major changes (order and higher, or major rearrangements of the taxonomic hierarchy). 
We anticipate that updates (new versions) will be published much more frequently than the time that has elapsed between PPG I and PPG II.
This will greatly benefit stakeholders who need an up-to-date, standardized taxonomy for ferns and lycophytes.

Another major change is that starting with PPG II, the taxonomy will include names at the level of species and below (PPG I treated only to genus).
This amount of data is too large to include in a traditional publication so it is omitted here, but it may be found online at our website or World Flora Online.
The large number of species, with ca. 14,000 accepted names, makes it infeasible to vote on every proposed change at the species level and below.
Therefore, we have established committees of specialists who are each responsible for maintaining taxonomic data at the species level and below for particular groups within pteridophytes (typically the genus or family level).
To minimize inconsistency among species classification, we aim to facilitate a discussions on the criteria applied by taxon experts to evaluate the confidence that the proposed taxonomic unit corresponds to a species.
The species data currently included in PPG II were included when the initial database was imported from World Ferns, and not all species names have been inspected by the PPG II community yet.
This work will continue indefinitely with the publication of PPG II.

## Classification

Circumscription and synonymy are not shown; this information can be found on the digital version of the classification at <https://pteridogroup.github.io>.
Species counts should be considered approximate.
All taxa are considered monophyletic unless otherwise mentioned.

```{r}
#| label: ppg-treatment
#| results: asis

children_taxa_count <- format_ppg_taxa_count(children_tally) |>
  assert(not_na, taxon_count) |>
  mutate(taxon_count = paste0(taxon_count, "."))

priority_sort <- set_taxon_priority(ppg, families_in_phy_order)

parent_taxon_count <-
  ppg |>
  dwc_to_tl(families_in_phy_order, return_taxlist = FALSE) |>
  count_taxon_parents() |>
  select(taxonID, n_parents) |>
  assert(not_na, everything()) |>
  assert(is_uniq, taxonID)

ppg_tl |>
  sort_taxa(priority = priority_sort) |>
  indented_list(print = FALSE) |>
  as_tibble() |>
  janitor::clean_names() |>
  select(
    taxonID = taxon_concept_id,
    scientificName = taxon_name,
    scientificNameAuthorship = author_name,
    taxonRank = level
  ) |>
  left_join(
    children_taxa_count,
    by = "taxonID",
    relationship = "one-to-one"
  ) |>
  left_join(
    parent_taxon_count,
    by = "taxonID",
    relationship = "one-to-one"
  ) |>
  left_join(
    select(
      ppg,
      taxonID,
      namePublishedIn
    ),
    by = "taxonID",
    relationship = "one-to-one"
  ) |>
  left_join(
    comments,
    by = "taxonID",
    relationship = "one-to-one"
  ) |>
  mutate(
    taxonRank_print = str_replace_all(taxonRank, "genus", "") |>
      str_to_sentence(),
    indent = rep_collapse("  ", n_parents),
    indent = paste0(indent, "* "),
    comment = tidyr::replace_na(comment, ""),
    taxon_count = tidyr::replace_na(taxon_count, "")
  ) |>
  mutate(
    # genera in bold italics, everything else in bold
    name_print = case_when(
      taxonRank == "genus" ~ glue("***{scientificName}***"),
      .default = glue("**{scientificName}**")
    ),
    pretty = glue(
      "{taxonRank_print} {name_print}　{scientificNameAuthorship}, {namePublishedIn}. {taxon_count} {comment}"
    ) |>
      as.character() |>
      str_replace_all("\\.+", ".") |>
      str_squish(),
    pretty = glue("{indent}{pretty}")
  ) |>
  select(pretty) |>
  pull(pretty) |>
  cat(sep = "\n")
```

## Tables

```{r}
#| label: tbl-taxa
#| tbl-cap:
#|   - "Number of taxa in PPG II vs. PPG I by taxonomic status
#|   (accepted or synonym). Complete synonymy was not presented in PPG I so
#|   this is not shown. PPG I did not treat nothotaxa."

ppg_2_taxa_count |>
  left_join(
    select(
      ppg_i_taxa_count,
      taxonRank,
      ppgi_accepted = accepted
    )
  ) |>
  filter(taxonRank != "tribe") |>
  mutate(
    taxonRank = str_to_sentence(taxonRank),
    across(where(is.integer), ~ scales::number(., big.mark = ","))
  ) |>
  select(
    Rank = taxonRank,
    `PPG II (accepted)` = accepted,
    `PPG II (synonym)` = synonym,
    `PPG I (accepted)` = ppgi_accepted
  ) |>
  knitr::kable()
```

```{r}
#| label: tbl-changes
#| tbl-cap:
#|   - "Count of taxonomic changes in PPG II vs. PPG I by rank, with
#|   the effect on the total number of taxa in each rank brought about by each
#|   change. Types of change include conserve: conserves one
#|   name over another; sink: subsumes one taxon into another; split: recognizes
#|   a new taxon; transfer: moves a taxon into a different position in the
#|   hierarchy."

ppg_issues_count |>
  filter(change != "none") |>
  mutate(rank = factor(rank, levels = c("Family", "Subfamily", "Genus", "Nothogenus"))) |>
  count(rank, change) |>
  mutate(effect = case_when(
    change %in% c("conserve", "expand", "shrink", "transfer") ~ "none",
    change == "sink" ~ "decrease",
    change == "split" ~ "increase",
  )) |>
  filter(
    change %in% c("conserve", "sink", "split", "transfer")
  ) |>
  mutate(
    change = str_to_sentence(change),
    effect = str_to_sentence(effect)
  ) |>
  arrange(rank, change) |>
  select(
    Rank = rank,
    Change = change,
    Effect = effect,
    n = n
  ) |>
  knitr::kable()
```

## Figures

```{r}
#| label: fig-tree
#| fig-width: 6.5
#| fig-height: 9
#| fig-cap:
#|   - "Summary cladogram showing relationships between families of
#|   tracheophytes recognized by PPG II. Topology for class Polypodiopsida
#|   based on Fern Tree of Life v`r ft_data_ver()`.
#| "

# Format family labels with genus/species counts. Keep num species for tip
# points
family_tip_labels <-
  ppg |>
  filter(taxonRank == "family", taxonomicStatus == "accepted") |>
  select(taxonID) |>
  left_join(children_tally, by = join_by(taxonID)) |>
  select(taxon_name, n_genera, n_species) |>
  mutate(
    family_label = glue::glue("{taxon_name} ({n_genera}/{n_species})") |>
      as.character()
  ) |>
  select(tip = taxon_name, family_label, n_species)

# Format clade labels for formal taxa
subclass_labels <- format_tip_labels("subclass", phy_family, ppg, ppg_tl)
order_labels <- format_tip_labels("order", phy_family, ppg, ppg_tl)
suborder_labels <- format_tip_labels("suborder", phy_family, ppg, ppg_tl)

# Format labels for informal clade names
informal_clades <- tribble(
  ~node, ~label,
  getMRCA(phy_family, c("Blechnaceae", "Polypodiaceae")), "eupolypods"
) |>
  mutate(label_type = "clade")

# Combine
all_labs <- subclass_labels |>
  bind_rows(order_labels) |>
  bind_rows(suborder_labels) |>
  bind_rows(informal_clades) |>
  rename(taxon = label)

# - right side clades
clade_labs_side <- all_labs |>
  filter(label_type == "clade") |>
  filter(
    taxon %in% c(
      "Polypodiineae",
      "Aspleniineae",
      "Lindsaeineae",
      "Cyatheales",
      "Salviniales",
      "Schizaeales"
    )
  ) |>
  # ggtree wants different column names for each 'label' in every dataset
  rename(taxon_clade = taxon)

# - internal nodes
clade_labs_node <- all_labs |>
  filter(label_type == "clade") |>
  filter(
    taxon %in% c(
      "eupolypods",
      "Polypodiales",
      "Polypodiidae",
      "Ophioglossidae"
    )
  ) |>
  rename(taxon_node = taxon)

# - right side tips
tip_labs_side <- all_labs |>
  filter(label_type == "tip") |>
  select(tip, taxon_tip = taxon)

# set font sizes
lab_font_size <- 2.5
tip_font_size <- 2.5
clade_lab_offset <- 8

# generate figure
tree_fig <- ggtree(phy_family, branch.length = "none") %<+%
  family_tip_labels %<+%
  clade_labs_node %<+%
  tip_labs_side +
  # Triangles scaled by species count
  geom_tippoint(
    aes(size = n_species),
    shape = 18, # triangle shape
    # x = triangle_offset,
    color = "black"
  ) +
  # Clade labels (nodes)
  geom_text_repel(
    aes(label = taxon_node),
    min.segment.length = 0,
    position = position_nudge_repel(x = -2.5, y = 1),
    size = lab_font_size
  ) +
  # Clade labels (right side)
  geom_cladelab(
    data = clade_labs_side,
    mapping = aes(node = node, label = taxon_clade),
    offset = clade_lab_offset,
    fontsize = lab_font_size,
    extend = 0.25
  ) +
  # Paraphyletic group labels (right side)
  geom_strip(
    "Gleicheniaceae",
    "Matoniaceae",
    label = "Gleicheniales",
    offset = clade_lab_offset,
    fontsize = lab_font_size,
    extend = 0.25,
    offset.text = 0.2
  ) +
  # Tip labels (family)
  geom_tiplab(
    aes(label = family_label),
    size = tip_font_size,
    offset = 0.45
  ) +
  # Tip labels (higher taxon)
  geom_tiplab(
    aes(label = taxon_tip),
    size = tip_font_size,
    offset = clade_lab_offset
  ) +
  xlim(-1, 33) +
  theme(
    legend.position = "none"
  )

ggsave(
  plot = tree_fig,
  width = 6.5,
  height = 8.5,
  units = "in",
  file = "fig_1.pdf"
)

tree_fig
```

## References
